### ISSUE TITLE
Navigation page: improve the Manage locations default option value

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#25313]
**Describe the bug**

In the new Navigation page, the "Manage locations" buttons opens a popover with a list of select elements. The default value of these selects (no selected value) is a hyphen:

<img width="295" alt="Screenshot 2020-09-14 at 19 39 40" src="https://user-images.githubusercontent.com/1682452/93119698-87c2cb80-f6c2-11ea-831e-cb6bb6d2f4b1.png">

In the current Menus screen instead, there's a more meaningful text "Select a Menu":

<img width="632" alt="Screenshot 2020-09-14 at 19 39 18" src="https://user-images.githubusercontent.com/1682452/93119741-9d37f580-f6c2-11ea-836c-d73ef1b6acb5.png">

The hyphen isn't meaningful and doesn't inform users very well. There's a lack of information that can be considered a bug. Any specific reason why a hyphen was used over the previous text? This doesn't seem a usability (and accessibility) improvement.

Also, when using a screen reader, hearing "hyphen, ticked" ("ticked" because it's the selected option) isn't great and potentially confusing.

**Expected behavior**
The default option to use a meaningful text.

**Editor version (please complete the following information):**
- WordPress version: 5.6 trunk
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? plugin
- If the Gutenberg plugin is installed, which version is it?  9.0.0-rc.1
